docs/reference/contributing/connectivity/LoRaPortingGuide.md

```diff
@@ -0,0 +1,71 @@
+### Porting LoRaWAN stack & LoRa RF drivers
+
+Mbed-OS contains a [native LoRaWAN stack (inside Mbed-OS tree)](https://github.com/ARMmbed/mbed-os/tree/master/features/lorawan) augmented with most commonly used [LoRa RF drivers (outside Mbed-OS tree)](https://github.com/ARMmbed/mbed-semtech-lora-rf-drivers). Arm encourages developers around the globe to contribute to any of the Mbed-OS features and functionality including LoRaWAN stack and related RF drivers using the general guidelines provided [here](https://os.mbed.com/docs/v5.8/reference/guidelines.html). 
+
+The information contained in this section can be classified in two sub-sections:
+
+1. Porting a LoRa RF driver for Arm Mbed LoRaWAN Stack or any other LoRaWAN stack if needed.
+
+2. Porting a 3rd party LoRaWAN stack to Mbed-OS 
+
+The idea is to achieve universal application portability, i.e., a single application is compatible with Arm Mbed LoRaWAN stack and any other 3rd party LoRaWAN stack harnessing Arm Mbed-OS. 
+
+#### Porting a LoRa RF driver 
+
+Arm Mbed-OS provides a generic API that serves as a template for any LoRa RF driver. [LoRaRadio](https://github.com/ARMmbed/mbed-os/blob/master/features/lorawan/LoRaRadio.h) is a pure virtual class and is an attempt to standardize the APIs across all LoRa radios. Any Mbed enabled LoRa radio driver implementation should present itself as a LoRaRadio. 
+
+<span class="images">![](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/lora_radio_inherit.png)<span>Figure 1. Existing Mbed LoRa RF drivers inherit from LoRaRadio class.</span></span>
+
+For a reference implementation, please check the existing [LoRa RF drivers](https://github.com/ARMmbed/mbed-semtech-lora-rf-drivers). Construction of a LoRaRadio object is a matter of taste. You may notice that the existing reference drivers allow construction of the LoRaRadio object with full pin definitions, that is to make sure that the driver is usable across platforms with any pin combination. You are free to use any form of construction as long as you provide a LoRaRadio object down to the Arm Mbed LoRaWAN stack. Utilization of an instance of `LoRaRadio` class for a 3rd party LoRaWAN stack is beyond the scope of this documentation but it should be trivial. 
+
+For API use cases, details, explanation and meaning please check the `LoRaRadio` class reference given below. Data structures provided in [LoRaRadio.h](https://github.com/ARMmbed/mbed-os/blob/master/features/lorawan/LoRaRadio.h) are carefully planned and designed. They carry most of the stuff you may ever need to write your LoRa RF driver with.
+
+Although implementation details are left for the developers to ponder on, there are a few recommendations:
+
+1. The driver should support both RTOS and non-RTOS, i.e., the driver should be able to defer interrupts in case of RTOS using any of the Mbed-OS synchronization methods.
+
```

Comment:
This is a recommendation that a driver should support both RTOS and non-RTOS. There are no tests to verify this. Drivers always come from outside the tree and it depends upon the user how he wishes to use the system as a whole. However, if the user wishes to contribute to official drivers repo, then the recommended way is to support both RTOS and non-RTOS.

Reply:
There is a need to have tests on the Mbed API for Lora to verify correct integration of those drivers. These drivers are *not* necessarily needed for those contributing with their code to Mbed OS, but also for those building applications/products - they need to have confidence that the porting has been done successfully.
For example, for the ESP8266 (external module), there are tests in the SAL/sockets layer - see example: https://os.mbed.com/docs/v5.8/reference/networkstack.html

Regardless of whether there are tests or not in place, the information in the doc is a bit short and should be expanded. Think about those people porting their device. What do they need to know?

Reply:
Green tea tests are currently internal. Please note that while you are reviewing this material, certain doxygen links are not visible. All the APIs are documented with all the necessary details in their respective classes. It will be repetitive to add that information once again.

Reply:
Product requirements have nothing to do with documentation. We are technically not yet ready to have same kind of tests. We will reach to that point later and we won't document anything which is not ready!